### DISCUSSION TITLE
Ligature with characters across different scripts (Latin + Greek)

### DISCUSSION THREAD

**Discussion by Revenantus556** [harfbuzz/harfbuzz#3524] [Q&A] [answered]
I'm working on a WOFF2 font using FontForge. All ligatures render correctly in Chrome, except those defined between θ and Latin vowels which always appear as separate characters, although the correct ligature does appear in the metrics preview in FontForge. 

If I do nothing other than change the character θ to a Latin character as a test, the ligatures appear as expected, so my working theory is that the issue is due to the fact that θ is the only character in my font defined by unicode as being in the Greek range whereas all other ligatures are between characters defined as Latin. Is it the case that characters in different scripts won't form ligatures?

StackExchange reference: https://stackoverflow.com/questions/71725387/browser-font-ligature-rendering-ligatures-between-%ce%b8-and-latin-characters-not-w

**Comment by khaledhosny** [comment, marked as answer]
Applications will split the text into different runs (single font, script, direction, language) before passing it to HarfBuzz, so the theta and the next Latin characters are never processed in a single run so can’t be ligated.

**Reply by Revenantus556** [reply]
Thanks very much for the confirmation. This would seem to mean I will have to avoid the use of the character θ which is unfortunate for proper IPA support, though admittedly this is a rather niche use case.

**Reply by khaledhosny** [reply]
It might be possible to make the case for Unicode to give theta Script_Extensions property of "Latn Grek" so at least implementations that use Script_Extensions for segmentation (I think Chrome does) can segment it with adjacent Latin text.

**Reply by Revenantus556** [reply]
This would solve this specific instance, thanks for the suggestion, since theta is the only non-Latn character relevant to the language I'm working on. More generally the IPA in its entirely includes further characters which aren't defined as Latn. I think there's a linguistic argument for being able to implement a general abugida purely with IPA characters, though this would require the entire IPA alphabet to be able to be processed as a single text run. Do you think there's scope for proposing this in some form and do you have any thoughts on how this would be proposed in a maintainable way? Adding Script_Extensions "Latn Grek" to all IPA characters? Adding a new script extension definition (e.g. "IPA")? While I'm confident the linguistic argument for being able to do it is there, I'm not able to evaluate what the appropriate method of handling this in unicode would be.

**Reply by khaledhosny** [reply]
I think the rest of Greek IPA characters have been encodes in Latin script (Latin beta, for instance, seems to have been encoded relatively recently), and there seem to have been several Unicode proposals for Latin theta, though I can't seem to find why it didn't get through. May be asking on Unicode mailing list would get you some clues.

**Reply by khaledhosny** [reply]
Script_Extensions seemed to be a lower effort option in case a dedicated code point was ruled out.

**Reply by Revenantus556** [reply]
You're correct; it looks like theta is indeed the single Greek character in the IPA which hasn't been given a separate codepoint. I've found two proposals to introduce a latin theta, though it doesn't appear that this has occurred yet. I'll enquire on the unicode mailing list as you suggest. Thanks very much for your help.

**Reply by dscorbett** [reply]
IPA isn’t the only system that mixes scripts. Other examples include UPA, Japanese, and Tamil-Grantha. A better general solution would be for the itemizer to keep multiple scripts together in the same item sometimes.

**Reply by jfkthame** [reply]
It's a bit unclear to me how this would interact with OpenType's organization of shaping as a set of features/lookups associated with a specific OpenType script tag. Given that (many of) the characters involved *do* have an association with specific scripts, the DFLT tag:

> A Script table with the script tag DFLT (default) may be used in a font to define features that are not script-specific. An application should use a DFLT script table if there is not a script table associated with the specific script of the text being formatted, or if the text does not have a specific script (for example, it contains only symbols or punctuation).

doesn't quite seem to be applicable here.
